src/scanner.c

```diff
@@ -0,0 +1,205 @@
+#include <assert.h>
+#include <stdio.h>
+#include <string.h>
+#include <tree_sitter/parser.h>
+
+#define MAX(a, b) ((a) > (b) ? (a) : (b))
+
+enum TokenType {
+    HERE_STRING_BODY,
+};
+
+typedef struct {
```

Comment:
consider write comment that points out the invariant of `String`: `cap == strlen(data) + 1`.
consider change all `uint32_t` to `size_t` as it can't work well on 64-bit machine.

Reply:
I don't understand the point of that

Reply:
My wrong. I misplaced this comment. See below.